### PR TITLE
Preserve LIMA_CIDATA_* variables when running a user provision script

### PR DESCRIPTION
**Pull request by frkd-dev**
### Description

[Lima's environment vars](https://github.com/lima-vm/lima/blob/master/pkg/cidata/cidata.TEMPLATE.d/lima.env) are not accessible from a user mode provision script like this:

```yaml
provision:
- mode: user
  script: |
    #!/bin/sh
    ln -s $LIMA_CIDATA_HOSTHOME_MOUNTPOINT/.conan ~/.conan
```

The above script creates a wrong link since the `LIMA_CIDATA_HOSTHOME_MOUNTPOINT` variable isn't visible from the script:
```
# ls -la ~
total 20
drwxr-xr-x    4 zzzzz    zzzzz         4096 Dec  4 15:01 .
drwxr-xr-x    3 root     root          4096 Dec  2 00:02 ..
-rw-------    1 zzzzz    zzzzz          234 Dec  4 11:58 .ash_history
lrwxrwxrwx    1 zzzzz    zzzzz            7 Dec  4 15:01 .conan -> /.conan
drwxr-xr-x    3 zzzzz    zzzzz         4096 Dec  4 11:54 .config
drwx------    2 zzzzz    zzzzz         4096 Dec  2 00:02 .ssh
```

To preserve LIMA_CIDATA_* variables in a user script the `sudo` command shall use `-E` option which preserves all environment variables.

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
`LIMA_CIDATA_*` vars are considered to be private to the internal script. I'm not sure we want to expose them.
At least, these will not be guaranteed to work.

**Comment by frkd-dev** [comment]
Sorry, but they are already exposed and used, for instance in [this template](https://github.com/lima-vm/lima/blob/master/examples/podman-rootful.yaml#L44). If this is not the intended behavior, then perhaps provision scripts should get the support of go template "actions" like `{{.Name}}`? Currently, they are unsupported, and writing provision scripts that rely on the info about VM (like in my case) is quite problematic.

**Comment by jandubois** [comment]
> `LIMA_CIDATA_*` vars are considered to be private to the internal script.

They are already exposed to system provisioning scripts, and sometimes needed there. I don't see any harm in exposing them to user provisioning scripts as well.

> I'm not sure we want to expose them. At least, these will not be guaranteed to work.

This is a good point. We should probably think about which variables we'll consider as an API, so provisioning scripts can rely on them, and which ones we consider internal, subject to change without warning.

Given that system provisioning scripts can already use these variables (and some of our templates do), we can't actually just change/drop variables because otherwise older machine instances may fail to work after a restart.

I'm going to approve this PR, but leave it to @AkihiroSuda to decide if it should be merged or not.

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Comment by jandubois** [comment]
I just noticed that the commit doesn't have a DCO signature, so that will need to be fixed before it can be merged: https://github.com/lima-vm/lima/pull/2052/checks?check_run_id=19294190163

**Comment by frkd-dev** [comment]
never used signatures before, so this is a good reason to start.

**Comment by jandubois** [comment]
> never used signatures before, so this is a good reason to start.

DCO signatures are required by all projects belonging to the CNCF: https://github.com/cncf/foundation/blob/main/dco-guidelines.md

**Comment by AkihiroSuda** [comment]
> perhaps provision scripts should get the support of go template "actions" like {{.Name}}?

👍 

This can be probably added in:
https://github.com/lima-vm/lima/blob/b453dedb757e5fc1e249f352d0812d008e66fa3f/pkg/store/instance.go#L233-L239

File: pkg/cidata/cidata.TEMPLATE.d/boot.sh
Comment (frkd-dev) [reply pkg/cidata/cidata.TEMPLATE.d/boot.sh:69]: Yeah, I agree, then this PR should be dropped as this is a dirty workaround.

**Comment by frkd-dev** [comment]
@AkihiroSuda if there are no plans or time from your side to add templates to provision scripts I could try to add this (if you have no objections).

**Comment by AkihiroSuda** [comment]
> @AkihiroSuda if there are no plans or time from your side to add templates to provision scripts I could try to add this (if you have no objections).

Thanks 👍

**Comment by jandubois** [comment]
I think this issue can be solved by executing user provisioning scripts with a small wrapper script that just reads `lima.env` the same way the `boot.sh` script does, and then `exec`s the actual user script.

I think I would prefer that mechanism over turning scripts into Go templates. I'll try to post a sample later today; just very busy right now.

**Comment by frkd-dev** [comment]
For my personal needs to run stuff on a local machine, using private vars is an acceptable solution even if they would change in the future. I cannot judge and argue somehow whether it's acceptable from the project's long-run perspective to expose them, so I leave this to maintainers. 😃

**Comment by jandubois** [comment]
> I think this issue can be solved by executing user provisioning scripts with a small wrapper script

Turns out this is a bit more complicated than anticipated because all the CIDATA files are only readable/executable by root:

```console
lima-alpine:~# ls -ld /mnt/lima-cidata/provision.user/
drwx------    3 root     dialout        336 Dec  5 21:26 /mnt/lima-cidata/provision.user/
```

So when I run the wrapper script with `sudo -i` then it can no longer exec the actual user script because it already lost access to it.

Out of time again; will have to think about it some more.

**Comment by jandubois** [comment]
> all the CIDATA files are only readable/executable by root

Looks like I was not paying proper attention because I was in a hurry. We already copy the user scripts into the user's home directory, so we can do the same for the launcher script. I've created #2056 to show what I mean.

I'm still not sure if turning scripts into Go templates has any advantages over simply exporting the variables. We probably have to keep supporting `lima.env` for backwards compatibility anyways, so adding templating would mostly be redundant (for system scripts).

@lima-vm/maintainers Let me know what you think!

**Pull request closed without merging**
